Dev to main

LeMaitre4523 opened the pull request.

camarm-dev committed:
[Fix] Utilisation de l'url du repo principal (de l'org `PapillonApp`) au lieu du fork

camarm-dev committed:
[Fix] Détails et précisions dans le guide utilisateur sur la documentation

- Arguments facultatifs
- Définitions manquantes pour `getMessaging.ts`

camarm-dev committed:
[Fix] Les arguments `message_id` ont été renommés `messageId`: utilisation des conventions de nommage javascript: _Camel Case_

camarm-dev committed:
fix: possibilité d'avoir les devoirs avec/sans les balises html dans `getHomeworks.ts` + contenu bien décodé (base64 -> string)

camarm-dev committed:
fix: ajout des dossiers de configuration des IDE (`.vscode/` et `.idea`) au `.gitignore`

camarm-dev committed:
fix: fin du programme si problème d'authentification

camarm-dev committed:
fix: fin du programme si problème d'authentification + message de connexion

camarm-dev committed:
add: exemple pour la timeline commune

camarm-dev committed:
fix: `getTimeline.ts`: les champs en base64 sont décodés !

camarm-dev committed:
add: exemple pour les devoirs

camarm-dev committed:
fix: linter fixes

camarm-dev committed:
add: update roadmap

camarm-dev committed:
fix: passage à la license `GPL` + modifications dans les attributions

camarm-dev committed:
add: fichier de citations

camarm-dev committed:
fix: update and renamed `citations.cff`

camarm-dev committed:
add: advice before committing in README

camarm-dev committed:
fix: désolé Vince 😅

camarm-dev committed:
fix: nouvelle license + script test fix

camarm-dev committed:
fix: nouvelle license + script test fix

camarm-dev committed:
add: tests unitaires

camarm-dev committed:
add: test action

camarm-dev committed:
fix: branche "stable" au lieur de "main" dans workflow

camarm-dev committed:
fix: ne pas lint les tests

camarm-dev committed:
fix: renvoie de `response.data` au lieu de la réponse complète dans `getHomeworks.ts` (+ modif doc et exemple en conséquences et pull src/types)

LeMaitre4523 committed:
add: workspace types

LeMaitre4523 committed:
feat: add getDiary, getTopics, getMembers of workspace

LeMaitre4523 committed:
Update roadmap

camarm-dev committed:
fix: amélioration de l'exemple EDT

camarm-dev committed:
add: `TODOS.md`

camarm-dev committed:
fix: utilisation particule du profil

camarm-dev committed:
fix: décodages des champs en base64 dans `getGrades.ts`

camarm-dev committed:
fix: création d'une fonction pour décoder la b64 (et gérer les problèmes d'accents)

camarm-dev committed:
fix: better typing for `getWorkspaces.ts`

camarm-dev committed:
fix: linter fixes

camarm-dev committed:
fix: récupération de la dernière version de `ecoledirecte-api-types`

camarm-dev committed:
fix: update de la documentation pour `getWorkspaces.ts`

camarm-dev committed:
fix: avertissement pour release et NPM

camarm-dev committed:
fix: fichiers générés dans `dist`

camarm-dev committed:
fix: exclude examples from file transpiling (`tsc`)

camarm-dev committed:
fix: added `dist` to .gitignore

camarm-dev committed:
fix: for npm publish, only upload 'dist' !

camarm-dev committed:
fix: download advice

camarm-dev committed:
fix: changed version: `0.2.1` + added changelog + changed build sy

camarm-dev committed:
fix: changed modula path

camarm-dev committed:
fix: created NPM org

camarm-dev committed:
fix: created NPM org

camarm-dev committed:
fix: npm organisation (finally !)

LeMaitre4523 committed:
edit function login(support mobile login) + add function renewToken

LeMaitre4523 committed:
Update version citation.cff

LeMaitre4523 committed:
Update version package.json (0.2.2)

camarm-dev reviewed (dismissed):
Nous avons fait du bon travail en vrai.

LeMaitre4523 committed:
Edit ed useragent and add error 526

LeMaitre4523 committed:
edit login function doc

LeMaitre4523 committed:
Add support mobile login function

LeMaitre4523 committed:
remove ua type

LeMaitre4523 committed:
add getEsidoc

LeMaitre4523 committed:
add getOrders

LeMaitre4523 committed:
add orders and esidoc to session

camarm-dev committed:
Update DOCUMENTATION.md

- Added anchor link to "Références"
- Properly renamed "EDCore"

camarm-dev committed:
fix: updated `types` & typed orders

camarm-dev committed:
fix: typed `auth.ts`; renewing token

camarm-dev committed:
fix: updated `types`

camarm-dev committed:
fix: deleted console.log

camarm-dev committed:
fix: enhanced documentation + fixed bad dev experience caused by login uuid; `v0.2.3`

camarm-dev reviewed (dismissed):
C'est bon pour moi, tout fonctionne.

camarm-dev committed:
add: new error

camarm-dev committed:
add: new error

camarm-dev committed:
add: ordering and deleting order

camarm-dev committed:
add: typed request body in `getDocuments.ts`

camarm-dev committed:
fix: removed school menu from orders

camarm-dev committed:
add: get school menus

camarm-dev committed:
fix: enhanced documentation

- Added guide for orders
- Started to add new modules
- Fixed cantine documentation

camarm-dev committed:
fix: linter

camarm-dev committed:
add: downloads

camarm-dev committed:
add: linked menus and downloads

camarm-dev committed:
fix: updated types

camarm-dev committed:
add: guide for downloads

camarm-dev committed:
fix: fileId can be a path (`string`) in cloud

camarm-dev committed:
add: documentation des nouvelles fonctionnalités

- Documentation `getOrders`
- Documentation `getEsidoc`
- Documentation `getDownloads`
- Documentation `auth.ts`

camarm-dev committed:
add: version `0.2.4`

camarm-dev reviewed (dismissed):
Tout est fonctionnel, on attend la review d'un coordinateur pour merge.

camarm-dev committed:
add: version `0.2.5`

- Ajout du téléchargement de fichier en base64

camarm-dev committed:
fix: branch is still "main", not "stable"

LeMaitre4523 committed:
Update CITATION.cff

camarm-dev committed:
fix: disabled tests (they are not finished)

camarm-dev committed:
Merge remote-tracking branch 'origin/dev' into dev

LeMaitre4523 committed:
Update DOCUMENTATION.md
